packages/react-i18n/src/babel-plugin/generate-dictionaries.ts

```diff
@@ -0,0 +1,74 @@
+import {join} from 'path';
+
+import fs from 'fs-extra';
+import cloneDeep from 'lodash.clonedeep';
+import merge from 'lodash.merge';
+
+import {DEFAULT_FALLBACK_LOCALE, findTranslationBuckets} from './shared';
+
+export interface Options {
+  fallbackLocale: string;
+  rootDir: string;
+}
+
+export async function generateTranslationDictionaries(
+  locales: string[],
+  {
+    fallbackLocale = DEFAULT_FALLBACK_LOCALE,
+    rootDir = process.cwd(),
+  }: Partial<Options> = {},
+) {
+  if (!locales || locales.length === 0) {
+    throw new Error(
+      'generateTranslationDictionary must be called with at least one locale.',
+    );
+  }
+
+  const translationBuckets = findTranslationBuckets(rootDir);
+
+  await Promise.all(
+    Object.entries(translationBuckets).map(
+      async ([translationsDir, translationFilePaths]) => {
+        const fallbackTranslations = await readLocaleTranslations(
+          fallbackLocale,
+          translationFilePaths,
+        );
+
+        const dictionary = await locales.reduce(async (accPromise, locale) => {
+          const localeTranslations = await readLocaleTranslations(
+            locale,
+            translationFilePaths,
+          );
+          const acc = await accPromise;
+          acc[locale] = merge(
```

Comment:
I guess we have to use lodash merge here because this needs to merge deeply and whatnot? Does make me wonder if we want to use `groupBy` from lodash for our bucketing operations to make them clearer, but /shrug

Reply:
Mmmmm, I'm conflicted about this because I *hate* adding lodash to anything 😸 The merge/clone things were unavoidable, but in for a penny...

I've created an issue to track this enhancement: https://github.com/Shopify/quilt/issues/1200